src/LightningUserWallet.ts

```diff
@@ -20,7 +21,7 @@ export class LightningUserWallet extends OnChainMixin(LightningMixin(UserWallet)
   async addEarn(ids) {
 
     if (this.user?.twilio?.carrier?.type === "voip") {
-      throw new Error("reward can only be given on non voip-based phone")
+      throw new CustomError("reward can only be given on non voip-based phone", "VOIP_REWARD_RESTRICTED", {forwardToClient: true, logger: this.logger, level: 'warn', metadata: undefined})
```

Comment:
I would recommend creating an enum with all your custom error messages so that you have a singular place to organize these error messages and update/reuse them.

`throw new CustomError(Errors.VOIP_BASED_PHONE,...) `

```
enum Errors {
  VOIP_BASED_PHONE = "reward can only be given on non voip-based phone",
  ...
}
```

Reply:
The mapping between error codes and error messages isn't 1 <-> 1. For the same error code, we sometimes have a different error message.

Reply:
I am suggesting that you have an enum to house the different string constants related to errors. 

This will allow the error objects to have many different types of error messages.

If you need a new text message, it can be a new item in the enum

```
throw new CustomError(Errors.VOIP_BASED_PHONE,...)
throw new CustomError(Errors.OTHER_ERROR,...)
```

```
enum Errors {
  VOIP_BASED_PHONE = "reward can only be given on non voip-based phone",
  ...
}
```

Reply:
I can see what you're suggesting here, however consolidating all the error messages in one place doesn't seem straightforward here. At the moment, some of our error messages are defined as default parameters in custom error classes, or they're scattered across the functions that throw them (this includes variable error messages).

I don't think it'll be a worthwhile improvement to move the errors from error.ts into an enum, while leaving the others as they are.